new tuist project does not run

### What happened?

I get the following error when running a new project: `is not a valid digit in integer literal` 

![Screenshot 2024-12-06 at 5 32 49 PM](https://github.com/user-attachments/assets/a01bf42f-4557-4650-8582-6e0e7c696594)

Previous Issue: https://github.com/tuist/tuist/issues/5224#issuecomment-2516353721

### How do we reproduce it?

**Minimal Reproduction Project**

https://github.com/navtoj/debug/tree/tuist-repro-1

**Manual Steps**

```
mkdir -p tuist-repro-1

cd tuist-repro-1

tuist init --platform macos --name 1Demo

tuist build
```

### Error log

<details>
<summary>
Press to Expand Code

[![tuist build](https://github.com/user-attachments/assets/29d95147-f4ea-4004-8408-cea17088d791)](https://github.com/tuist/tuist/issues/7144)

</summary>

```sh
tuist build

Loading and constructing the graph
It might take a while if the cache is empty
Using cache binaries for the following targets: 
Generating workspace 1Demo.xcworkspace
Generating project 1Demo
Building scheme 1Demo
[1Demo] Write Auxiliary File 1Demo.hmap
[1Demo] Write Auxiliary File 1Demo-project-headers.hmap
[1Demo] Write Auxiliary File 1Demo-own-target-headers.hmap
[1Demo] Write Auxiliary File 1Demo.SwiftFileList
[1Demo] Write Auxiliary File 1Demo-OutputFileMap.json
[1Demo] Write Auxiliary File 1Demo_const_extract_protocols.json
[1Demo] Write Auxiliary File 1Demo.SwiftConstValuesFileList
[1Demo] Write Auxiliary File Entitlements.plist
[1Demo] Write Auxiliary File 1Demo.LinkFileList
[1Demo] Write Auxiliary File 1Demo.DependencyMetadataFileList
[1Demo] Write Auxiliary File 1Demo-generated-files.hmap
[1Demo] Write Auxiliary File 1Demo-all-target-headers.hmap
[1Demo] Write Auxiliary File 1Demo-all-non-framework-target-headers.hmap
[1Demo] Write Auxiliary File 1Demo-DebugDylibInstallName-normal-arm64.txt
❌ /Users/navtoj/Developer/tuist-repro-1/Derived/Sources/TuistAssets+1Demo.swift:23:14: 'D' is not a valid digit in integer literal

❌ /Users/navtoj/Developer/tuist-repro-1/Derived/Sources/TuistAssets+1Demo.swift:23:13: enum name can only start with a letter or underscore, not a number

❌ /Users/navtoj/Developer/tuist-repro-1/Derived/Sources/TuistAssets+1Demo.swift:25:36: 'D' is not a valid digit in integer literal

❌ /Users/navtoj/Developer/tuist-repro-1/Derived/Sources/TuistAssets+1Demo.swift:34:21: 'D' is not a valid digit in integer literal

❌ /Users/navtoj/Developer/tuist-repro-1/Derived/Sources/TuistAssets+1Demo.swift:34:20: class name can only start with a letter or underscore, not a number

❌ /Users/navtoj/Developer/tuist-repro-1/Derived/Sources/TuistAssets+1Demo.swift:63:19: 'D' is not a valid digit in integer literal

❌ /Users/navtoj/Developer/tuist-repro-1/Derived/Sources/TuistAssets+1Demo.swift:63:18: expected type name in extension declaration

❌ /Users/navtoj/Developer/tuist-repro-1/Derived/Sources/TuistAssets+1Demo.swift:63:18: expected '{' in extension

❌ /Users/navtoj/Developer/tuist-repro-1/Derived/Sources/TuistAssets+1Demo.swift:65:29: 'D' is not a valid digit in integer literal

❌ /Users/navtoj/Developer/tuist-repro-1/Derived/Sources/TuistAssets+1Demo.swift:80:16: 'D' is not a valid digit in integer literal

❌ /Users/navtoj/Developer/tuist-repro-1/Derived/Sources/TuistAssets+1Demo.swift:25:46: consecutive declarations on a line must be separated by ';'

❌ /Users/navtoj/Developer/tuist-repro-1/Derived/Sources/TuistAssets+1Demo.swift:25:46: expected declaration

❌ /Users/navtoj/Developer/tuist-repro-1/Derived/Sources/TuistAssets+1Demo.swift:80:15: expected parameter type following ':'

[1Demo] Compiling ContentView.swift
[1Demo] Compiling GeneratedAssetSymbols.swift
[1Demo] Compiling _1DemoApp.swift
[1Demo] Compiling TuistBundle+_1Demo.swift
[1Demo] Compiling TuistAssets+1Demo.swift
The 'xcodebuild' command exited with error code 65 and message:
** BUILD FAILED **


The following build commands failed:
	SwiftEmitModule normal arm64 Emitting\ module\ for\ _Demo (in target '1Demo' from project '1Demo')
	EmitSwiftModule normal arm64 (in target '1Demo' from project '1Demo')
	Building workspace 1Demo with scheme 1Demo
(3 failures)

Consider creating an issue using the following link: https://github.com/tuist/tuist/issues/new/choose
```

</details>

### macOS version

14.7.1 (23H222)

### Tuist version

4.36.0

### Xcode version

16.1 (16B40)

--- a/cli/Sources/TuistGenerator/Mappers/SynthesizedResourceInterfaceProjectMapper.swift
+++ b/cli/Sources/TuistGenerator/Mappers/SynthesizedResourceInterfaceProjectMapper.swift
@@ -292,7 +292,7 @@ public struct SynthesizedResourceInterfaceProjectMapper: ProjectMapping { // swi
         if paths.isEmpty {
             renderedInterfaces = []
         } else {
-            let name = target.name.camelized.uppercasingFirst
+            let name = target.name.toValidSwiftIdentifier()
             renderedInterfaces = [
                 (
                     "Tuist\(templateName)+\(name)",
@@ -300,7 +300,7 @@ public struct SynthesizedResourceInterfaceProjectMapper: ProjectMapping { // swi
                         parser: resourceSynthesizer.parser,
                         parserOptions: resourceSynthesizer.parserOptions,
                         templateString: templateString,
-                        name: target.productName.camelized.uppercasingFirst,
+                        name: target.productName.toValidSwiftIdentifier(),
                         bundleName: project.options.disableBundleAccessors ? nil : "Bundle.module",
                         paths: paths
                     )

--- a/cli/Tests/TuistGeneratorTests/ProjectMappers/SynthesizedResourceInterfaceProjectMapperTests.swift
+++ b/cli/Tests/TuistGeneratorTests/ProjectMappers/SynthesizedResourceInterfaceProjectMapperTests.swift
@@ -876,6 +876,61 @@ final class SynthesizedResourceInterfaceProjectMapperTests: TuistUnitTestCase {
         XCTAssertTrue(plistNames.value.contains("XML.plist"))
     }
 
+    func testMap_usesValidSwiftIdentifiersForGeneratedResourceInterfaces() async throws {
+        // Given
+        let names = ThreadSafe<[String]>([])
+        synthesizedResourceInterfacesGenerator.renderStub = { _, _, _, name, _, _ in
+            names.mutate { $0.append(name) }
+            return ""
+        }
+        let projectPath = try temporaryPath()
+        let targetPath = projectPath.appending(component: "1Demo")
+        let assetCatalog = targetPath.appending(component: "Assets.xcassets")
+        try await fileSystem.makeDirectory(at: assetCatalog)
+        try await fileSystem.touch(assetCatalog.appending(component: "Contents.json"))
+
+        let target = Target.test(
+            name: "1Demo",
+            resources: .init([.folderReference(path: assetCatalog)])
+        )
+        let project = Project.test(
+            path: projectPath,
+            targets: [target],
+            resourceSynthesizers: [
+                .init(
+                    parser: .assets,
+                    parserOptions: [:],
+                    extensions: ["xcassets"],
+                    template: .defaultTemplate("Assets")
+                ),
+            ]
+        )
+
+        // When
+        let (mappedProject, sideEffects) = try await subject.map(project: project)
+
+        // Then
+        XCTAssertEqual(names.value, ["_1Demo"])
+        XCTAssertEqual(
+            sideEffects,
+            [
+                .file(
+                    FileDescriptor(
+                        path: projectPath
+                            .appending(component: Constants.DerivedDirectory.name)
+                            .appending(component: Constants.DerivedDirectory.sources)
+                            .appending(component: "TuistAssets+_1Demo.swift"),
+                        contents: Data()
+                    )
+                ),
+            ]
+        )
+        XCTAssertEqual(
+            mappedProject.targets[target.name]?.sources.first?.path.basename,
+            "TuistAssets+_1Demo.swift"
+        )
+    }
+
     // MARK: - Helpers
 
     private func stub(file: AbsolutePath) async throws {

